pkg/container/docker_run.go

```diff
@@ -125,6 +126,13 @@ func (cr *containerReference) Start(attach bool) common.Executor {
 				cr.attach().IfBool(attach),
 				cr.start(),
 				cr.wait().IfBool(attach),
+				cr.tryReadUID(),
+				cr.tryReadGID(),
+				func(ctx context.Context) error {
+					// If this fails, then folders have wrong permissions on non root container
+					_ = cr.Exec([]string{"chown", "-R", fmt.Sprintf("%d:%d", cr.UID, cr.GID), cr.input.WorkingDir}, nil, "0", "")(ctx)
+					return nil
```

Comment:
these chown operations cause a significant performance issue for large repos (e.g. git-lfs), especially because they likely use the --bind option

Can you evaluate alternative mechanisms to check for incorrect permissions? or otherwise make this logic smart so it only executes for non-root containers?

Reply:
I agree that we could skip chown, if cr.UID and cr.GID is 0 ( root ). You can open a Pullrequest adding an if around the chown commands added here.
> Can you evaluate alternative mechanisms to check for incorrect permissions?

Do you know one? I didn't found a way to tell docker to apply the owner of the tar archive to folders. Obviously if the container has a tar tool installed, then we can use it instead of using docker cp.